pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -324,8 +326,10 @@ func (p *LaunchTemplateProvider) getAL2UserData(ctx context.Context, constraints
 	var userData bytes.Buffer
 	userData.WriteString(fmt.Sprintf(`#!/bin/bash -xe
 exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+%s
 /etc/eks/bootstrap.sh '%s' %s \
     --apiserver-endpoint '%s'`,
+		p.ReplaceENIMaxPodsFile(ctx, constraints),
```

Comment:
I worry that we're gonna run headlong into userdata limits with this approach.

Reply:
We could double the size of our user-data today and still be within limits. 

We could also only populate the instance type options that we are going to launch. I'm doing all of them here for cache efficiency for the LT. Since we clean the LTs up now, it might be less of a big deal, other than the small bit of latency creating the LT.